### DISCUSSION TITLE
ww3_multi.inp?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [ufs-community/ufs-weather-model#805] [Q&A] [answered]
Could someone point me to documentation on how ww3_multi.inp is generated and what the time entries mean? Specifically I'm looking to understand lines like this:
line 17: 20130401 000000   10800  20130403 000000  1
The YYYYMMDD sections are presumably the start and end dates for the runs, but I'm not sure what to make of the 10800. Similarly:
line 641: 20130403 030000   3024000  20130403 030000
appears near the end of the file.

Thanks!

**Comment by JessicaMeixner-NOAA** [comment, marked as answer]
@benjamin-cash the line numbers are not helpful as you can have as many comment lines as you want. 

There are example input files here: 
https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_multi.inp
that provide more documentation.  

The 1 at the end of the line unfortunately is not yet documented in that file.  I'm assuming that's for one of the output, likely gridded output which means you are outputting the file per time step.  If you send the full file I'd be happy to explain them further, but in general those lines are: 

TimeStart DT dt TimeEnd 1 

Where Time start is a YYYYMMDD HHMMSS time format, for starting the output and the time end of the output is the same format.  DT is the tilmestep in seconds for the output and the 1 means write each file individually instead of all together.

**Reply by benjamin-cash** [reply]
@JessicaMeixner-NOAA Thanks! The specific file I am looking at is from your run directory here:
/scratch2/NCEPDEV/climate/Jessica.Meixner/p7workflowupdate/test01/DATA/test01/2013040100/gfs/fcst.124782

Since that was a shorter test I need to update it for the 35-day run I want to make, along with the other files (ice_in, etc.). I'm pretty sure I've worked out the changes I need to make for the other files but ww3_multi.inp is a little opaque.

**Reply by JessicaMeixner-NOAA** [reply]
So the second date which corresponds to the end date just needs to be updated for your 35 day run instead of a 2 day run.  The workflow takes care of this here: https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/scripts/exgfs_wave_prep.sh#L819

**Reply by benjamin-cash** [reply]
Where do the 03000 and 3024000 come from? Those are the two that I'm least certain of.

**Reply by DeniseWorthen** [reply]
3024000/86400 = 35

**Reply by benjamin-cash** [reply]
Thanks @DeniseWorthen ! And the 03000?

**Reply by JessicaMeixner-NOAA** [reply]
That's the HHMMSS of the end time. 

20130403 030000 3024000 20130403 030000
YYYYMMDD HHMMSS(start)  DT(in seconds) YYYYMMDD HHMMSS(end)

**Reply by benjamin-cash** [reply]
I'm confused about why that 3024000 is coming from 35 days, since this is a 2-day test? And why is HHMMSS 000000 at the beginning of the file but 030000 at the end?

**Reply by JessicaMeixner-NOAA** [reply]
DT is in seconds so 35 days * 24 hours/day * 60 min/hour * 60 seconds/min = 3024000s so 35 days is  3024000s, and that is the chosen time step for restart, which yes is longer than the total runtime, so in this case no restarts are written.    The variables are chosen independently of each other in the workflow.

**Reply by benjamin-cash** [reply]
Got it. And the difference between the HHMMSS in those two spots?

**Reply by JessicaMeixner-NOAA** [reply]
It's the HHMMSS of the start time and the end time of that output.

**Reply by benjamin-cash** [reply]
One more question (for now :) - Right above the line for the time step for the restart there is this:
$ Keep next two lines formatting as is to allow proper parsing
   20130402 000000   0  20130402 000000 T

Those dates aren't the start or end date, so I'm not sure where they come from.

**Reply by JessicaMeixner-NOAA** [reply]
@benjamin-cash the 0 for the dt means that no output (this particular being output for the first stream of restarts w/out a date stamp in the name) for this will not be written.  Again, to better understand how this file is created the best place to look is where this is created in the workflow, which starts here: https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/scripts/exgfs_wave_prep.sh#L819

**Reply by benjamin-cash** [reply]
@JessicaMeixner-NOAA I have looked at that section of the code and I'm afraid that it hasn't helped me all that much. I can see there are a large number of variable substitutions, but unless I am just not seeing something there is nothing in that code to tell me what the variables mean or where they were substituted in the file. I spent some time trying to find $PARMwave/ww3_multi.${NET}.inp.tmpl, which seems like it should have that information, but couldn't find anything that matched that name format.

**Reply by JessicaMeixner-NOAA** [reply]
The $PARMwave/ww3_multi.${NET}.inp.tmpl is here: https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/parm/wave/ww3_multi.gfs.inp.tmpl 

Most of the parameters are defined either here:  https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/workflow/cases/prototype7.yaml#L71-L77 or here: https://github.com/NOAA-EMC/global-workflow/blob/feature/coupled-crow/workflow/defaults/wave.yaml

**Reply by benjamin-cash** [reply]
Ah, thanks! Ok, so it looks like the settings that I am still trying to understand are:
RST_BEG   DTRST  RST_END

From the code you pointed me to, it looks like RST_BEG is substituted as: $time_rst_ini. That variable seems to come from: $ymdh_rst_ini, which depends on ${RSTOFFSET}. ${RSTOFFSET} looks to depend on a number of other variables, some of which are coming out of a set of if-then statements and other dependencies that I don't think I have enough information to untangle. I think I may just need to get a sample set of 35-day run files in order to answer this.

**Reply by JessicaMeixner-NOAA** [reply]
You don't actually need a restart file from WW3 unless you are restarting something, so you can just set DT=0 for both and wet the YYYYMMDD to whatever you want. @jiandewang sent you a 35 day run input file.
